templates/parachain/node/src/service.rs

```diff
@@ -202,8 +202,32 @@ fn start_consensus(
 		client.clone(),
 	);
 
+	let client_for_closure = client.clone();
 	let params = AuraParams {
-		create_inherent_data_providers: move |_, ()| async move { Ok(()) },
+		create_inherent_data_providers: Arc::new(move |parent, _| {
+			let slot_duration = sc_consensus_aura::standalone::slot_duration_at(
+				client_for_closure.as_ref(),
+				parent,
+			);
+			let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
+			let slot = slot_duration.map(|slot_duration| {
+				sp_consensus_aura::inherents::InherentDataProvider::from_timestamp_and_slot_duration(
+					*timestamp,
+					slot_duration,
+				)
+			});
+			async move { Ok((slot?, timestamp)) }
+		})
```

Comment:
It appears to me that this is used in multiple places, but I might miss subtle differences between them. It would be nice to dedup this if possible.

Reply:
It reads nicer to me if we don't repeat same type in the long form :), but that's why I wanted to have a second pass over the PR, since I haven't followed precisely the reasoning here: https://github.com/paritytech/polkadot-sdk/pull/8446#issuecomment-2900732981. The principle sounds correct to me, but I will revisit this soon.

Reply:
I still think it makes sense to dedup/contain the closures and reuse it, unless @bkchr initial comment points to something else we miss.